### PR TITLE
Support specifying options for port and path

### PR DESCRIPTION
**Pull request by trygveaa**
This adds support for specifying which port and path to use, or if memory should be used instead of a path, in `serverless.yml`.

It also removes the compiled files from git, and changes the type for the server to be more correct.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,15 +33,23 @@ Install the plugin with `yarn add --dev serverless-dynalite-local`. Add
...
 
 ```typescript
 const client = new DynamoDBClient({
   endpoint: "http://localhost:4567",
 });
 ```
 
+The following options are supported in `custom.dynalite` in `serverless.yml`:
+
+| Option | Description | Type | Default value |
```
Comment (hawkaa) [inline README.md:47]: Thank you!! I knew this was going to be useful one day but I decided to skip it for the first version.

File: src/index.ts [hunk 2]
```diff
@@ -13,15 +14,15 @@ class ServerlessDynaliteLocal {
   // Logging function, stolen from the serverless object
   log: (str: string, pluginName: string) => void;
 
-  // Hard coded port for dynalite
+  // Default port for dynalite
   port = 4567;
-  path = "./dynalite";
+  path: string | undefined = "./dynalite";
```
Comment (hawkaa) [inline src/index.ts:19]: Why does this need to be `... | undefined`?

File: src/index.ts [hunk 3]
```diff
@@ -33,18 +34,29 @@ class ServerlessDynaliteLocal {
...
+    if (serverless.service.custom.dynalite?.port != null) {
+      this.port = serverless.service.custom.dynalite.port;
+    }
+
+    if (serverless.service.custom.dynalite?.path != null) {
+      this.path = serverless.service.custom.dynalite.path;
+    }
+
+    if (serverless.service.custom.dynalite?.inMemory) {
+      this.path = undefined;
```
Comment (hawkaa) [inline src/index.ts:47]: Never mind

File: src/index.ts [hunk 4]
```diff
@@ -33,18 +34,29 @@ class ServerlessDynaliteLocal {
...
   }
 
   async startHandler() {
     const server = dynalite({ path: this.path });
-    await new Promise<void>((resolve, reject) =>
-      server.listen(this.port, (err: string | undefined) =>
-        err ? reject(err) : resolve()
-      )
+    await new Promise<void>((resolve) =>
+      server.listen(this.port, () => resolve())
```
Comment (hawkaa) [inline src/index.ts:54]: does this reject the promise if the server start fails? What was there before is my default "turn a callback into a promise" boilerplate.

File: types/dynalite/index.d.ts [hunk 5]
```diff
@@ -1,11 +1,8 @@
 declare module "dynalite" {
-  type DynaliteServer = {
-    listen: (port: number, callback: (err?: string) => void) => void;
-    close: (callback: (err?: string) => void) => void;
-  };
+  import { Server } from "http";
```
Comment (hawkaa) [inline types/dynalite/index.d.ts:2]: Thank you. So much nicer!!

**Review by hawkaa — APPROVED** [review]
Thank you @trygveaa !

File: src/index.ts (hunk 4 above)
Comment (trygveaa) [reply src/index.ts:54]: As far as I can see from the [node documentation](https://nodejs.org/docs/latest-v14.x/api/net.html#net_event_listening) and the types, and confirmed by testing, `listen` does not pass any parameters to the callback, so with the previous code, `reject` would never have been called.

If you want to handle errors, you instead have to call `server.on('error', err => ...)` before calling `listen`. But if you don't do this, the error is thrown as an unhandled exception instead, which I think is fine in this case, so therefore we don't need any error handler.

**Pull request merged**